dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -51,6 +51,178 @@ public static void prepare() throws Exception {
+
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
+        this.addPermission(role, host, true);
+
+        //Saving and testing GroupCondition
+        rulesAPI.saveConditionGroup(conditionGroup, user, false);
+
+        List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        assertEquals(1, allRules.size());
+
+        final Rule ruleFromDataBase = allRules.get(0);
+
+        final List<ConditionGroup> groups = ruleFromDataBase.getGroups();
+        assertEquals(1, groups.size());
+
+        assertEquals(groups.get(0).getOperator(), conditionGroup.getOperator());
+
+        //Saving and testing Condition
+        final Condition condition = new ConditionDataGen().group(groups.get(0)).next();
+
+        rulesAPI.saveCondition(condition, user, false);
+
+        allRules = rulesAPI.getAllRules(user, false);
+        assertEquals(1, allRules.size());
+        final List<Condition> conditions = allRules.get(0).getGroups().get(0).getConditions();
+
+        assertEquals(1, conditions.size());
+
+        assertEquals(condition.getOperator(), conditions.get(0).getOperator());
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveConditionGroup(ConditionGroup, User, boolean)}
+     * When: User without permission try to create a Group Condition
+     * Should: Throw a DotSecurityException
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldNotSaveCondition() throws DotDataException, DotSecurityException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule = new RuleDataGen().host(host).nextPersisted();
+
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().rule(rule).next();
+        rulesAPI.saveConditionGroup(conditionGroup, user, false);
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User with permission try to create a Rule's Action
+     * Should: Save it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveRuleAction() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule = new RuleDataGen().host(host).nextPersisted();
+
+        this.addPermission(role, host, true);
+
+        final RuleAction ruleAction = new RuleActionDataGen().rule(rule).next();
+
+        rulesAPI.saveRuleAction(ruleAction, user, false);
+
+        final List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        assertEquals(1, allRules.size());
```

Comment:
assert not needed

Reply:
done https://github.com/dotCMS/core/pull/18127/commits/d8a32ec5a775f434c7ee34caeaa3b0e19018ab5f#diff-9baded4fb0fb63e2648d9c9c4380ec1aL142